src/game/client/components/render_layer.cpp

```diff
@@ -1059,6 +1165,21 @@ bool CRenderLayerQuads::DoRender(const CRenderLayerParams &Params) const
 	// skip rendering if detail layers if not wanted
 	if(m_Flags & LAYERFLAG_DETAIL && !g_Config.m_GfxHighDetail && Params.m_RenderType != CMapLayers::TYPE_FULL_DESIGN) // detail but no details
 		return false;
+
+	if(m_QuadRenderGroup.m_Clipped)
+	{
+		CUIRect Screen;
+		Graphics()->GetScreen(&Screen.x, &Screen.y, &Screen.w, &Screen.h);
```

Comment:
```suggestion
		float ScreenX0, ScreenY0, ScreenX1, ScreenY1;
		Graphics()->GetScreen(&ScreenX0, &ScreenY0, &ScreenX1, &ScreenY1);
```

This otherwise this reads weird because there are `ScreenWidth` and `Screen.w` variables, but `Screen.w` isn't the width but the position of the corner.

Seems to be copied from another location in `maplayers.cpp` but that should also be avoided.